source/common.ts

```diff
@@ -474,19 +258,34 @@ const i18n = {
 i18n.__ = i18n.getTranslation;
 i18n.addTranslation = i18n.addTranslations;
 
-function format(integer: number, separator: string) {
-  let result = '';
-  while (integer) {
-    const n = integer % 1e3;
-    integer = Math.floor(integer / 1e3);
-    if (integer === 0) {
-      return n + result;
+function interpolateTranslation(
+  variables: Omit<GetTranslationOptions, "_locale">,
+  translation: string,
+  open: string,
+  close: string
+) {
+  let interpolatedTranslation = translation;
+  Object.entries(variables).forEach(([key, value]) => {
+    const tag = open + key + close;
+    if (interpolatedTranslation.includes(tag)) {
+      interpolatedTranslation = interpolatedTranslation.split(tag).join(value as string);
     }
+  });
+  return interpolatedTranslation;
+}
 
-    result = separator + (n < 10 ? '00' : n < 100 ? '0' : '') + n + result;
-  }
-
-  return '0';
+function normalizeGetTranslation(locales: string[], key: string, hideMissing: boolean) {
+  let translation: unknown;
+  locales.some(locale =>
+    i18n
+      ._normalizeWithAncestors(locale)
+      .some(
+        locale => (translation = get(i18n._translations, `${locale}.${key}`)),
+      ),
+  );
+  const translationWithHideMissing = translation ? `${translation}` : hideMissing ? '' : key;
+
+  return translationWithHideMissing;
 }
```

Comment:
1. These two functions should be part of the `i18n` object.
2. Options like `open`, `close`, and `hideMissing` should be taken directly from options.
3. The type of `variables` should be `Record<string, unknown>`. Right now there's a problem, that the `GetTranslationOptions` contains `_namespace` but it's not used in `getTranslation`.
4. I'd say that `normalizeGetTranslation` is not behaving as expected, as if the parent locale (e.g., `en`) specifies a translation and the child locale (e.g., `en-us`) specifies an empty string (`''`) or zero (`0`), the latter is ignored. I _think_ we should make it differentiate between `undefined` and any other value.
    * Additionally, we could make it significantly faster, as right now we're splitting `key` into parts just to join it later and then pass it to `get` that does the same. However, we can do it later.